operator: Avoid spamming logs with entire identity object

lvyanru8200 opened the pull request:
Signed-off-by: yanru.lv <yanru.lv@daocloud.io>

Please ensure your pull request adheres to the following guidelines:

- [ ] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [ ] All code is covered by unit and/or runtime tests where feasible.
- [ ] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [ ] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [ ] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [ ] Provide a title or release-note blurb suitable for the release notes.
- [ ] Thanks for contributing!

<!-- Description of change -->

Fixes: https://github.com/cilium/cilium/issues/21900

nebril reviewed (commented):
Thanks for the PR!

AFAIU the UID will be printed currently with other info from identity, is there any particular reason to limit the information printed?

christarazi reviewed (changes requested):
Thanks for the PR. Change looks good. Could you follow step 5 of https://docs.cilium.io/en/v1.12/contributing/development/contributing_guide/#submit-pr and reformat the commit msg accordingly?  Please also share the justification within the commit msg for why this change is needed.

joestringer commented on operator/cmd/k8s_identity.go:
```diff
@@ -117,7 +117,7 @@ func identityGCIteration(ctx context.Context, clientset k8sClient.Clientset) {
 					identity.Annotations = make(map[string]string)
 				}
-				log.WithField(logfields.Identity, identity).Info("Marking identity for later deletion")
+				log.WithField(logfields.Identity, identity.UID).Info("Marking identity for later deletion")
```
`logfields.Identity` usually refers to the identity number, which is probably `identity.Name` or similar?

The UID could still potentially be interesting in the case of multiple of the same-named resource, so maybe it can be extended to use `log.WithFields()` and include both the `logfields.Identity` as the identity number, and also `logfields.K8sUID` for the UID.

lvyanru8200 committed:
operator: Avoid spamming logs with entire identity object

Printing the entire structure of the identity in the log makes the log hard to read, so only the uid of the identity object is printed

Fixes: #21900

Signed-off-by: yanru.lv <yanru.lv@daocloud.io>

christarazi reviewed (commented):
Please squash your commits into one commit.
